src/Razor/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorProjectEngine.cs

```diff
@@ -94,7 +94,8 @@ protected override RazorCodeDocument CreateCodeDocumentCore(RazorProjectItem pro
             IReadOnlyList<RazorSourceDocument> importSourceDocuments = null, 
             IReadOnlyList<TagHelperDescriptor> tagHelpers = null,
             Action<RazorParserOptionsBuilder> configureParser = null,
-            Action<RazorCodeGenerationOptionsBuilder> configureCodeGeneration = null)
+            Action<RazorCodeGenerationOptionsBuilder> configureCodeGeneration = null,
+            string cssScope = null)
```

Comment:
Is it valuable for the css scope to be settable / an input? Wondering if this is better off acting more like a checksum where it's an agreed upon hashing mechanism of various inputs.

Reply:
The main thing is that only some, not all, of the components should emit scope attributes. 

If the compiler knew which ones then it could duplicate the hashing logic itself. However since we have to communicate which ones need the scope, I thought we might as well also avoid the need to duplicate logic for computing the scope value.

Reply:
> The main thing is that only some, not all, of the components should emit scope attributes.

What's the criteria for if a component should emit scope attributes?

Reply:
Whether or not the developer chooses to declare scoped styles for it, by having a sibling CSS file (eg MyComponent.razor.css).

Reply:
To be clear, I know we *could* have the Razor compiler look on disk for .css files, but that seems to violate the idea that this sort of thing is controlled and overridable in MSBuild.

Additionally, there’s a use case for the exact format of the scope identifiers (eg length) to be controllable by developers through csproj properties. Passing the identifiers in from MSBuild means we can do this pretty trivially and without needing more parameters and logic inside the Razor compiler.

Reply:
> To be clear, I know we _could_ have the Razor compiler look on disk for .css files, but that seems to violate the idea that this sort of thing is controlled and overridable in MSBuild.

Ya I definitely wouldn't want that either. I'm more trying to land on if we're going to need these css files specifically in the future. For instance, we track "import" documents today aka _ViewImports.cshtml, _Imports.razor etc. by declaring them as inputs into the system because they impact the way the top-level Razor file is parsed. If the answer in the future is "we don't need them as an input for any other reason" then this is a totally fine solution. 

So do you envision us needing these files or their filepaths in the future?

Reply:
I don't have any reason to envision rzc needing access to the contents of the CSS files. I was hoping to keep it as agnostic as possible to CSS concepts. Just giving it the minimal information needed to emit some extra specific attributes for specified components hopefully avoids coupling it to any more detailed knowledge of CSS.